crates/archive/src/tarball.rs

```diff
@@ -117,9 +114,14 @@ impl Archive for Tarball {
 /// downloading the entire gzip file. For very small files it's unlikely to be
 /// more efficient than simply downloading the entire file up front.
 fn fetch_isize(url: &str, len: u64) -> Result<[u8; 4], ArchiveError> {
-    let range_header = Range::Bytes(vec![ByteRangeSpec::FromTo(len - 4, len - 1)]);
+    let mut header_values = Vec::with_capacity(1);
+    Range::bytes(len - 4..=len - 1)
```

Comment:
I see why you kept this, as it's the same range as it was previously, but given it's doing `..=` now, I think it would be clearer to drop the `-1` and use `..` instead:

```suggestion
    Range::bytes(len - 4..len)
```